src/res_core.ml

```diff
@@ -640,23 +642,24 @@ let rec parseLident p =
   let startPos = p.Parser.startPos in
   match p.Parser.token with
   | Lident ident ->
-    Parser.next p;
+    Parser.next p [@doesNotRaise];
     let loc = mkLoc startPos p.prevEndPos in
     (ident, loc)
+  | Eof ->
+    ("_", mkLoc startPos p.prevEndPos)
   | _ ->
-    begin match recoverLident p with
+    begin match recoverLidentNoEof p with
     | Some () ->
       parseLident p
     | None ->
       ("_", mkLoc startPos p.prevEndPos)
     end
 
-let parseIdent ~msg ~startPos p =
+let parseIdentNoNext ~msg ~startPos p =
```

Comment:
We do `Parser.next` after calling this function.
This is because the caller has more context on what the token can be.